src/scala/scripts/TwitterScroogeGenerator.scala

```diff
@@ -126,37 +119,27 @@ object ScroogeGenerator {
 
     val scrooge = new Compiler
 
-    // we need to extract into the same tree, as that is the only way to get relative imports between them working..
-    // AS SUCH, we are just going to try extracting EVERYTHING to the same tree, and we will just error if there
-    // are more than one.
-    val _tmp =  Files.createTempDirectory(tmp, "jar")
-    // This will only be meaningful if they have absolute_prefix set
-    scrooge.includePaths += _tmp.toString
-
-    def extract(jars: Set[Path]): Set[Path] =
-      jars.flatMap { jar =>
-        val files = extractJarTo(jar, _tmp)
-        files.foreach { scrooge.includePaths += _.toString }
-        files
-      }
-
-    val immediateThriftSrcs = extract(immediateThriftSrcJars)
+    // these are the ones we are actually generating
+    def allFilesInZips(i: Iterable[Path]): Set[String] =
+      i.flatMap(listJar(_)).toSet
 
-    immediateThriftSrcs.foreach { scrooge.thriftFiles += _.toString }
-
-    val onlyTransitiveThriftSrcs = extract(onlyTransitiveThriftSrcJars)
-
-    val intersect = onlyTransitiveThriftSrcs.intersect(immediateThriftSrcs)
+    val immediateThriftSrcs = allFilesInZips(immediateThriftSrcJars)
+    immediateThriftSrcs.foreach { scrooge.thriftFiles += _ }
+    immediateThriftSrcJars.foreach { path => scrooge.includePaths += path.toString }
```

Comment:
well, the srcs are just the files in the zip that we need to build. I'll comment.